src/Components/pages/LoginwCustomHook.js

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import Logo from "../../Assets/work.webp";
+import LogoSmall from "../../Assets/bigoh-logo-1.webp";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import { toast } from "react-toastify";
+import useCustomForm from "../../hook/useCustomForm";
+
+// Validation function
+const validateForm = (input) => {
+  const errors = {};
+  if (!input.email) {
+    errors.email = "Email is required";
+  } else if (!/\S+@\S+\.\S+/.test(input.email)) {
+    errors.email = "Email address is invalid";
+  }
+  if (!input.password) {
+    errors.password = "Password is required ";
+  }
+  if (
+    !(
+      input.password.length >= 8 &&
+      /[a-zA-Z]/.test(input.password) &&
+      /\d/.test(input.password)
+    )
+  ) {
+    errors.password =
+      "Password should be at least 8 characters long and alphanumeric";
+  }
+
+  return errors;
+};
+
+const Login = () => {
+  const { input, register, handleSubmit, errors } = useCustomForm(
+    { email: "", password: "" },
+    validateForm
+  );
+
+  const handleLogin = async (data) => {
+    try {
+      const response = await axios.post(
+        `${process.env.REACT_APP_API}user/login`,
+        data
+      );
+      console.log(response);
+      toast.success(response.data.message);
+      localStorage.setItem("userInfo", response.data.token);
```

Comment:
constant variable for the the id of localStorage. const LocalStoargeKeys = { userInfo: 'userInfor' }

Reply:
Done